bin/git-commit-proxy-version

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -eu
+
+if [ $# -ne 1 ]; then
+    echo "usage: $(basename $0) PROXY-REVISION" >&2
+    exit 64
+fi
+new_proxy_revision="$1"
+
+bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+rootdir="$( cd $bindir/.. && pwd )"
+
+old_proxy_version="$(tr -d '\n' <"$rootdir/.proxy-version")"
+if [ -z "$old_proxy_version" ]; then
+    echo "missing .proxy-version" >&2
+    exit 1;
+fi
+
+# Checkout the linkerd2-proxy repo to resolve the new proxy version to a SHA
+# and obtain the commit log since the prior version.
+tmp=$(mktemp -d -t l2-proxy.XXX)
+git clone --depth=100 https://github.com/linkerd/linkerd2-proxy "$tmp"
```

Comment:
It seems pretty unlikely, but i suppose this _could_ fail weirdly if the new revision is over 100 commits old? I doubt that will be a common occurance though.

Reply:
Yeah... don't think we need to worry about being particularly robust. We could try to rev-parse the old sha to be sure...